src/clients/c++/perf_client/perf_client.cc

```diff
@@ -555,6 +749,28 @@ main(int argc, char** argv)
     Usage(argv, "zero input can't be set when data directory is provided");
   }
 
+
+  if (using_concurrency_range && using_old_options) {
+    Usage(argv, "can not use deprecated options with concurrency_range");
+  } else if (using_old_options) {
+    if (dynamic_concurrency_mode) {
+      concurrency_range[CONCURRENCY_RANGE::kEND] = max_concurrency;
+    }
+    concurrency_range[CONCURRENCY_RANGE::kSTART] = concurrent_request_count;
+  }
+
+  if (concurrency_range[CONCURRENCY_RANGE::kEND] == NO_LIMIT &&
+      latency_threshold_ms == NO_LIMIT) {
+    Usage(
+        argv,
+        "The end of the search range and the latency limit can not be both 0 "
+        "simultaneously");
+  }
+
+  if ((!url_specified) && protocol == perfclient::ProtocolType::GRPC) {
```

Comment:
You don't need parens around the !url_specified but you should have parens around the "protocol == ...". In general always put parens are conditional sub-expressions to avoid any precedence ambiguity.

Reply:
fixed